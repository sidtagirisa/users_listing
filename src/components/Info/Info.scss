@import "../../styles/variables.scss";

.information {
  margin-bottom: 5%;
}

.heading {
  margin: auto;
  background: $theme1;
  border-radius: 5px;
  color: white;
  padding: 5px 10px;
  font-weight: 600;
  font-size: 1.2rem;
  text-transform: uppercase;
}

.value {
  margin-top: 3%;
  margin-left: 5%;
}

.description {
  margin-left: 5%;
}