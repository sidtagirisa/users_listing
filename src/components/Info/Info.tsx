import React from 'react';
import './Info.scss';

export default function Info(props: {
  heading: string;
  value: string | JSX.Element;
  description?: string;
}): JSX.Element {
  return (
    <div className="information">
      <h4 className="heading">{props.heading}</h4>
      <p className="value">{props.value}</p>
      {props.description ? (
        <p className="description">{`${props.description}`}</p>
      ) : (
        <></>
      )}
    </div>
  );
}
