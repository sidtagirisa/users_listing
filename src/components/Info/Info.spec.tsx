import { render } from '@testing-library/react';
import React from 'react';
import Info from '.';

describe('# Info', () => {
  it('should render with info passed', async () => {
    const { container } = render(
      <Info heading="some value" value="some value" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with description if passed', async () => {
    const { container } = render(
      <Info
        heading="some value"
        value="some value"
        description="some description"
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
