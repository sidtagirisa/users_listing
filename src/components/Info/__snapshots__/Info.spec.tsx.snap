// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Info should render with description if passed 1`] = `
<div
  class="information"
>
  <h4
    class="heading"
  >
    some value
  </h4>
  <p
    class="value"
  >
    some value
  </p>
  <p
    class="description"
  >
    some description
  </p>
</div>
`;

exports[`# Info should render with info passed 1`] = `
<div
  class="information"
>
  <h4
    class="heading"
  >
    some value
  </h4>
  <p
    class="value"
  >
    some value
  </p>
</div>
`;
